tests/DecisionServiceTests/DecisionServiceTest.php

```diff
@@ -603,6 +614,445 @@ public function testGetVariationWithBucketingId()
 
         $variationKey = $optlyObject->getVariation($this->experimentKey, $userId, $userAttributesWithBucketingId);
         $this->assertEquals($this->variationKeyControl, $variationKey, sprintf('Variation "%s" does not match expected user profile variation "%s".', $variationKey, $this->variationKeyControl));
+    }
+    
+     //should return nil and log a message when the feature flag's experiment ids array is empty
+    public function testGetVariationForFeatureExperimentGivenNullExperimentIds(){
+
+        $feature_flag = $this->config->getFeatureFlagFromKey('empty_feature');
+        $this->loggerMock->expects($this->at(0))
+        ->method('log')
+        ->with(Logger::DEBUG, "The feature flag 'empty_feature' is not used in any experiments.");
+
+        $this->assertSame(
+            $this->decisionService->getVariationForFeatureExperiment($feature_flag,'user1',[]),
+            null
+        );
+    }
+
+     //should return nil and log when the experiment is not in the datafile 
+    public function testGetVariationForFeatureExperimentGivenExperimentNotInDataFile(){
+
+        $boolean_feature = $this->config->getFeatureFlagFromKey('boolean_feature');
+        $feature_flag = clone $boolean_feature;
+        $feature_flag->setExperimentIds(["29039203"]);
+
+        $this->loggerMock->expects($this->at(0))
+        ->method('log')
+        ->with(Logger::ERROR, 'Experiment ID "29039203" is not in datafile.');
+
+        $this->loggerMock->expects($this->at(1))
+        ->method('log')
+        ->with(Logger::INFO, 
+            "The user 'user1' is not bucketed into any of the experiments on the feature 'boolean_feature'.");
+
+        $this->assertSame(
+            $this->decisionService->getVariationForFeatureExperiment($feature_flag,'user1',[]),
+            null
+        );
+    }
+
+    //should return nil and log when the user is not bucketed into the feature flag's experiments
+    public function testGetVariationForFeatureExperimentGivenNonMutexGroupAndUserNotBucketed(){
+        $multivariate_experiment = $this->config->getExperimentFromKey('test_experiment_multivariate');
+        $map = [ [$multivariate_experiment, 'user1', [], null] ];
+
+        //make sure the user is not bucketed into the feature experiment
+        $this->decisionServiceMock->expects($this->at(0))
+        ->method('getVariation')
+        ->will($this->returnValueMap($map));
```

Comment:
why it should be map?

Reply:
getVariation returns null, you are including user,  and other stuff in the $map and returning null?

Reply:
$map = [ [$multivariate_experiment, 'user1', [], null] ];
This means that when getVariation will be called with ($multivariate_experiment, 'user1', []) and should return the null.

Reply:
This could be done without map. Since getVariation will only be called once in this method